### PR TITLE
MISUV-4736: CL257 - Update error messages for Agents in Tax year summary page

### PR DESCRIPTION
**Pull request by AshokGitAccount**

### CODE REVIEW

**Comment by avelx** [comment]
@AshokGitAccount  - I would believe you would need a simple tests to verify that all resources you added under C15~ exists; just a simple test to make sure key/value pair exists for Eng/Cy would work fine;

File: conf/messages [hunk 1]
```diff
@@ -272,6 +272,76 @@ tax-year-summary.message.C559105                                   = the amount
...
+tax-year-summary.agent.message.C15015                              = your client has claimed to carry forward a loss to set against general income of the next year. They also need to make the claim in the same year the loss arose.
+tax-year-summary.agent.message.C15016                              = your client uses cash basis accounting. This means that they cannot claim to set losses against other taxable income.
+tax-year-summary.agent.message.C15102                              = the total amount of one-off Gift Aid payments is more than the total Gift Aid payments your client made.
+tax-year-summary.agent.message.C15103                              = Gift Aid payments made this year treated as paid in the previous year are more than the total Gift Aid payments your client made.
+tax-year-summary.agent.message.C15104                              = the value of qualifying investments your client gifted to non-UK charities are more than the value of these items they gifted to charity:<br><ul class="govuk-list govuk-list--bullet"> <li> qualifying shares and securities </li> <li> qualifying land and buildings </li> </ul>
```
Comment (avelx) [inline conf/messages:283]: this might be not a good idea to keep HTML formatting as a part of resource file, we have views for this

Comment (AshokGitAccount) [reply conf/messages:283]: It's been implemented for the individual so I have implemented the same for Agent

Comment (avelx) [reply conf/messages:283]: no html and styles in the message files pls

File: app/controllers/agent/errors/asd.sc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import scala.util.matching.Regex
```
Comment (avelx) [inline app/controllers/agent/errors/asd.sc:1]: Pls drop this file from the PR

Comment (avelx) [reply app/controllers/agent/errors/asd.sc:1]: Why do we need this regexp btw ???

File: app/views/TaxYearSummary.scala.html [hunk 3]
```diff
@@ -48,17 +48,24 @@
...
-        @errorMessage(model)
+@getMessageKey(isAgent: Boolean) = {
+  @if(isAgent) {tax-year-summary.agent.message} else {tax-year-summary.message}
+}
+
+@taxCalculationOverview(model: TaxYearSummaryViewModel, isAgent: Boolean) = {
+    @if(model.messages.exists(_.errorMessages.size > 0)) {
+       @{
+            val key = getMessageKey(isAgent)
+            errorMessage(model,s"${key}".trim(), isAgent)
```
Comment (avelx) [inline app/views/TaxYearSummary.scala.html:67]: you can move trim in getMessageKey function,

Comment (AshokGitAccount) [reply app/views/TaxYearSummary.scala.html:67]: I've removed the function and used a function literal instead. We dont need the trim() now.

File: app/controllers/agent/errors/asd.sc (hunk 2 above)
Comment (AshokGitAccount) [reply app/controllers/agent/errors/asd.sc]: Deleted file

**Review by avelx — APPROVED** [review]
approved

**Pull request merged**